core/src/main/scala/filodb.core/memstore/IndexBootstrapper.scala

```diff
@@ -48,6 +47,36 @@ class IndexBootstrapper(colStore: ColumnStore) {
       }
   }
 
+  /**
+   * Same as bootstrapIndexRaw, except that we parallelize lucene update for
+   * faster bootstrap of large number of index entries in downsample cluster.
+   * Not doing this in raw cluster since parallel TimeSeriesPartition
+   * creation requires more careful contention analysis
+   */
+  def bootstrapIndexDownsample(index: PartKeyLuceneIndex,
+                     shardNum: Int,
+                     ref: DatasetRef)
+                    (assignPartId: PartKeyRecord => Int): Task[Long] = {
+
+    val recoverIndexLatency = Kamon.gauge("shard-recover-index-latency", MeasurementUnit.time.milliseconds)
+      .withTag("dataset", ref.dataset)
+      .withTag("shard", shardNum)
+    val start = System.currentTimeMillis()
+    colStore.scanPartKeys(ref, shardNum)
+      .mapAsync(Runtime.getRuntime.availableProcessors()) { pk =>
```

Comment:
Should it be a config based instead which will help us in testing with different value?

Reply:
I didn't want config overload. Better to cross that bridge later if really needed. This default seems ok.